src/main/kotlin/no/nav/helse/spion/vedtaksmelding/VedtaksmeldingProcessor.kt

```diff
@@ -1,41 +1,23 @@
 package no.nav.helse.spion.vedtaksmelding
 
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.Dispatchers
-import no.nav.helse.spion.vedtaksmelding.failed.FailedVedtaksmelding
-import no.nav.helse.spion.vedtaksmelding.failed.FailedVedtaksmeldingRepository
-import no.nav.helse.utils.RecurringJob
-import java.time.Duration
-import java.util.*
+import com.fasterxml.jackson.databind.ObjectMapper
+import no.nav.helse.arbeidsgiver.bakgrunnsjobb.BakgrunnsjobbProsesserer
+import java.time.LocalDateTime
 
 class VedtaksmeldingProcessor(
-        private val kafkaVedtaksProvider: VedtaksmeldingProvider,
-        private val service: VedtaksmeldingService,
-        private val failedVedtaksmeldingRepository: FailedVedtaksmeldingRepository,
-        coroutineScope: CoroutineScope = CoroutineScope(Dispatchers.IO),
-        waitTimeWhenEmptyQueue: Duration = Duration.ofSeconds(30)
-) : RecurringJob(coroutineScope, waitTimeWhenEmptyQueue) {
+        val vedtaksmeldingService: VedtaksmeldingService,
+        val om: ObjectMapper
+) : BakgrunnsjobbProsesserer {
 
-    override fun doJob() {
-        do {
-            val wasEmpty = kafkaVedtaksProvider
-                    .getMessagesToProcess()
-                    .onEach { tryProcessOneMessage(it) }
-                    .isEmpty()
+    companion object {
+        val JOBB_TYPE = "vedtaksmelding"
+    }
 
-            if (!wasEmpty) {
-                kafkaVedtaksProvider.confirmProcessingDone()
-            }
-        } while (!wasEmpty)
+    override fun prosesser(jobbData: String) {
+        vedtaksmeldingService.processAndSaveMessage(om.readValue(jobbData, SpleisMelding::class.java))
     }
 
-    private fun tryProcessOneMessage(melding: SpleisMelding) {
-        try {
-            service.processAndSaveMessage(melding)
-        } catch (t: Throwable) {
-            val errorId = UUID.randomUUID()
-            logger.error("Feilet vedtaksmelding, Database ID: $errorId", t)
-            failedVedtaksmeldingRepository.save(FailedVedtaksmelding(melding, t.message, errorId))
-        }
+    override fun nesteForsoek(forsoek: Int, forrigeForsoek: LocalDateTime): LocalDateTime {
+        return LocalDateTime.now().plusHours((forsoek * forsoek).toLong())
```

Comment:
Kanskje vi skal tweake denne så det forsøkes litt oftere enn det her. Det tar 16 dager før en melding feiler permanent gitt 10 forsøk med denne backoff'en.

Vet at jeg la inn denne i spinn, men tenker nå at det kanskje er for lang tid før vi ta affære. Kanskje vi kan ta det opp med Kine og høre hva som er fornuftig per ting